### DISCUSSION TITLE
[BUG] New-PnPAzureCertificate: Object reference not set to an instance of an object with Powershell 7.4.1

### DISCUSSION THREAD

**Discussion by OASISfr** [pnp/powershell#3838] [General]
**PnP.Powershell version :** 2.3.0 / 2.4.0
**OS :** Windows 11
**Powershell version :** 7.4.1

**Reporting Issue :**

New-PnPAzureCertificate -OutPfx pnp.pfx -OutCert pnp.cer -CertificatePassword (ConvertTo-SecureString -String "pass@word1" -AsPlainText -Force)

Generate an error : Object reference not set to an instance of an object.

Commandlet is operational with Powershell 7.4.0

**EDIT 14/03/2024 :** 
**Powershell version IS NOT the problem.**
**The commandlet work with PnP.Powershell 2.2.0 / Powershell 7.4.1**

**Get-Error :**

Exception             :
    Type       : System.NullReferenceException
    TargetSite :
        Name          : WriteAzureCertificateOutput
        DeclaringType : PnP.PowerShell.Commands.Base.GetPnPAdalCertificate
        MemberType    : Method
        Module        : PnP.PowerShell.dll
    Message    : Object reference not set to an instance of an object.
    Source     : PnP.PowerShell
    HResult    : -2147467261
    StackTrace :
   at PnP.PowerShell.Commands.Base.GetPnPAdalCertificate.WriteAzureCertificateOutput(PSCmdlet cmdlet, X509Certificate2 certificate, SecureString password) in
c:\build\src\Commands\Base\GetAzureCertificate.cs:line 82
   at PnP.PowerShell.Commands.Base.NewPnPAdalCertificate.ProcessRecord() in c:\build\src\Commands\Base\NewAzureCertificate.cs:line 104
   at System.Management.Automation.CommandProcessor.ProcessRecord()
CategoryInfo          : NotSpecified: (:) [New-PnPAzureCertificate], NullReferenceException
FullyQualifiedErrorId : System.NullReferenceException,PnP.PowerShell.Commands.Base.NewPnPAdalCertificate
InvocationInfo        :
    MyCommand        : New-PnPAzureCertificate
    ScriptLineNumber : 1
    OffsetInLine     : 2
    HistoryId        : 4
    Line             :  New-PnPAzureCertificate -OutPfx pnp.pfx -OutCert pnp.cer -CertificatePassword (ConvertTo-SecureString -String "pass@word1" -AsPlainText -Force)
    Statement        : New-PnPAzureCertificate -OutPfx pnp.pfx -OutCert pnp.cer -CertificatePassword (ConvertTo-SecureString -String "pass@word1" -AsPlainText -Force)
    PositionMessage  : At line:1 char:2
                       +  New-PnPAzureCertificate -OutPfx pnp.pfx -OutCert pnp.cer -Certificat …
                       +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : New-PnPAzureCertificate
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1

**Comment by jackpoz** [comment]
Cannot reproduce the issue using v2.4.8 on Windows 10
![image](https://github.com/pnp/powershell/assets/1153754/f7d39c4b-1f57-47a7-a14f-6b50355f568a)

Did the command create the pfx and cer files ? According to the stacktrace you posted, it failed when showing the certificate in the console but after it got created.

**Reply by OASISfr** [reply]
Hello.

I'am disapointed to see my bug turn in discussion !!

Yes, the command create the file.

**Reply by jackpoz** [reply]
Could you please create a brand new certificate pair and upload both pfx and cer ? Make sure not to use it anywhere.
The code https://github.com/pnp/powershell/blob/e25326f7ebef6bf48a431897619372db5dd49bb9/src/Commands/Base/GetAzureCertificate.cs#L89-L95 does a few things and any of them could cause the NullReferenceException exception. 11d195b6b9fe8150fd403e928a38a477e8a27cc7 added some extra code related to San names, which is what most likely is causing the issue in the 2.3.0+ versions

**Comment by dpvq** [comment]
Having the same problem using PnP.PowerShell 3.1.0

```
$cert = New-PnPAzureCertificate -CommonName "cert" -OutPfx .\cert.pfx -OutCert .\cert.cer -ValidYears 2 -CertificatePassword  (ConvertTo-SecureString -String "xxxxxxxxxxxxxxx" -AsPlainText -Force)
New-PnPAzureCertificate: Object reference not set to an instance of an object.
```

**Comment by dpvq** [comment]
Seem a duplicate of https://github.com/pnp/powershell/issues/5130 and possible fix https://github.com/pnp/powershell/pull/5137

**Reply by reshmee011** [reply]
Thanks so much @dpvq for linking both issues together. Hopefully #5137 is the fix if the issue is related to using Non-English locale.
